Source/WebCore/platform/mediastream/RealtimeMediaSourceCapabilities.h

```diff
@@ -114,17 +114,30 @@ class CapabilityValueOrRange {
 template<class Encoder>
 void CapabilityValueOrRange::encode(Encoder& encoder) const
 {
-    encoder.encodeFixedLengthData(reinterpret_cast<const uint8_t*>(&m_minOrValue), sizeof(ValueUnion), alignof(ValueUnion));
-    encoder.encodeFixedLengthData(reinterpret_cast<const uint8_t*>(&m_max), sizeof(ValueUnion), alignof(ValueUnion));
+    encoder.encodeObject(m_minOrValue);
+    encoder.encodeObject(m_max);
     encoder << m_type;
 }
 
 template<class Decoder>
 bool CapabilityValueOrRange::decode(Decoder& decoder, CapabilityValueOrRange& valueOrRange)
 {
-    return decoder.decodeFixedLengthData(reinterpret_cast<uint8_t*>(&valueOrRange.m_minOrValue), sizeof(ValueUnion), alignof(ValueUnion))
-        && decoder.decodeFixedLengthData(reinterpret_cast<uint8_t*>(&valueOrRange.m_max), sizeof(ValueUnion), alignof(ValueUnion))
-        && decoder.decode(valueOrRange.m_type);
+    auto minOrValue = decoder.template decodeObject<ValueUnion>();
+    if (!minOrValue)
+        return false;
+
+    auto max = decoder.template decodeObject<ValueUnion>();
+    if (!max)
+        return false;
+
+    auto type = decoder.template decode<Type>();
+    if (!type)
+        return false;
+
```

Comment:
I think applying batched, branch-free decoding will be simplest (or already is simplest) in the generated serializers code.

Decoding from an already-invalid decoder perhaps deserves some investigation. The buffer position would be null, and I believe any decoding would fail early in the buffer containment check, but I didn't review thoroughly.